File: leetcodeContest/187周/4.cpp
class Solution {
public:
    int kthSmallest(vector<vector<int>>& mat, int k) {
        int m=mat.size();
        int n=mat[0].size();
        vector<int> res(mat[0]);
        for(int i=1;i<m;++i)
        {
            multiset<int> st;
            for(int x:res)
            {
                for(int y:mat[i])
                {
                    st.insert(x+y);
                }
            }
            res.assign(st.begin(),st.end());
            res.resize(min(k,(int)res.size()));
        }
        return res[k-1];
    }
};

File: leetcodeContest/187周/4.py
class Solution:
    def kthSmallest(self, mat: List[List[int]], k: int) -> int:
        m, n = len(mat), len(mat[0])
        H = [0]
        for l in mat:
            HH = []
            for c in l:
                for h in H:
                    HH.append(c + h)
            H = sorted(HH)[:k]
        return H[-1]
